### PR TITLE
Add gRPC Logging

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/8221

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
provider: gRPC requests and response logging have been enabled
```

### CODE REVIEW

**Comment by upodroid** [comment]
Bump

File: mmv1/third_party/terraform/utils/config.go.erb [hunk 1]
```diff
@@ -77,6 +83,79 @@ type providerMeta struct {
...
+	LogFormat string
+}
+
+// Borrowed logic from https://github.com/sirupsen/logrus/blob/master/json_formatter.go and https://github.com/t-tomalak/logrus-easy-formatter/blob/master/formatter.go
+func (f *Formatter) Format(entry *logrus.Entry) ([]byte, error) {
+	if !logging.IsDebugOrHigher() {
+		return nil, nil // Suppress logs if TF_LOG is not DEBUG or TRACE
+	}
+	output := f.LogFormat
+	entry.Level = logrus.DebugLevel // Force Entries to be Debug
```
Comment (c2thorn) [inline mmv1/third_party/terraform/utils/config.go.erb:97]: Do you believe it's worth switching to `TraceLevel` for when TF_LOG is `TRACE` ?

Comment (upodroid) [reply mmv1/third_party/terraform/utils/config.go.erb:97]: No, debug is verbose enough. Also, we are stripping out grpc internals log anyway so no benefit there.

**Comment by upodroid** [comment]
Fixed go.sum conflicts, this is ready to be merged

File: mmv1/third_party/terraform/utils/config.go.erb [hunk 2]
```diff
@@ -801,8 +900,9 @@ func (c *Config) NewCloudIdentityClient(userAgent string) *cloudidentity.Service
 
 func (c *Config) BigTableClientFactory(userAgent string) *BigtableClientFactory {
 	bigtableClientFactory := &BigtableClientFactory{
-		UserAgent:           userAgent,
-		TokenSource:         c.tokenSource,
+		UserAgent:  		 userAgent,
+		TokenSource:		 c.tokenSource,
+		gRPCLoggingOptions:  c.gRPCLoggingOptions,
```
Comment (c2thorn) [inline mmv1/third_party/terraform/utils/config.go.erb:905]:
```suggestion
		UserAgent:                    userAgent,
		TokenSource:                c.tokenSource,
		gRPCLoggingOptions:  c.gRPCLoggingOptions,
```

File: mmv1/third_party/terraform/utils/config.go.erb [hunk 3]
```diff
@@ -801,8 +900,9 @@ func (c *Config) NewCloudIdentityClient(userAgent string) *cloudidentity.Service
 
 func (c *Config) BigTableClientFactory(userAgent string) *BigtableClientFactory {
 	bigtableClientFactory := &BigtableClientFactory{
-		UserAgent:           userAgent,
-		TokenSource:         c.tokenSource,
+		UserAgent:                    userAgent,
+		TokenSource:                c.tokenSource,
+		gRPCLoggingOptions:  c.gRPCLoggingOptions,
```
Comment (c2thorn) [inline mmv1/third_party/terraform/utils/config.go.erb:905]:
```suggestion
		UserAgent:           userAgent,
		TokenSource:         c.tokenSource,
		gRPCLoggingOptions:  c.gRPCLoggingOptions,
```

**Review by c2thorn — APPROVED** [review]

**Pull request merged**